### PR TITLE
implemented SSSDK

### PR DESCRIPTION
**Pull request by Asheshp23**

### CODE REVIEW

File: Simple Salesforce SDK/.gitignore [hunk 1]
```diff
@@ -1,9 +0,0 @@
-.DS_Store
-/.build
-/Packages
-/*.xcodeproj
-xcuserdata/
-DerivedData/
-.swiftpm/config/registries.json
-.swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata
-.netrc
```
Comment (ipranay) [inline Simple Salesforce SDK/.gitignore:9]: We should use this gitignore, no?

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,4 @@ DerivedData/
 .swiftpm/config/registries.json
 .swiftpm/xcode/package.xcworkspace/contents.xcworkspacedata
 .netrc
+.DS_Store
```
Comment (ipranay) [inline .gitignore]: This file already has DS Store ignored at the top

File: Simple Salesforce SDK/.gitignore (hunk 1 above)
Comment (Asheshp23) [reply Simple Salesforce SDK/.gitignore:9]: yes.

File: README.md [hunk 3]
```diff
@@ -51,4 +55,87 @@ Fetches raw data and returns JSON,
...
+1. Open your project in Xcode
+
+2. Click "File" -> "Swift Packages" -> "Add Package Dependency..."
+
+3. Paste the following URL: https://github.com/deveronuas/sssdk.git
+
+4. Click "Next" -> "Next" -> "Finish"
+
+## General Usage
+import SwiftUI
```
Comment (ipranay) [inline README.md]: Please use a swift code block

```suggestion

```swift
import SwiftUI
```

File: README.md [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # Simple Salesforce SDK
+SSSDK is a simple framework that can be used as a replacement of the Salesforce's connected mobile SDK application.SSSDK supports iOS 15 or above(SwiftUI).
```
Comment (ipranay) [inline README.md]:
```suggestion
This is a simplified SDK that can be used as a replacement for Salesforce's connected mobile SDK in your SwiftUI application. It supports iOS 15 or above.
```

File: README.md [hunk 5]
```diff
@@ -18,7 +19,8 @@ SSSDK.shared.configure(host:, redirectUri:, clientId:, clientSecret:)
 ```
 
 ### login
-Shows a login popup with the configured host
+Returns login view with the configured host and redirectUri 
+(for more information please visit the following link -> https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_user_agent_flow.htm&type=5)
```
Comment (ipranay) [inline README.md]: If you would like to refer docs, maybe better if refer Salesforce docs?

File: README.md [hunk 6]
```diff
@@ -36,7 +38,9 @@ SSSDK.shared.logout()
 ```
 
 ### refershAccessToken
-Refreshes the `access_token`, then updates the keychain. It also uses Salesforce's introspection endpoint to fetch and store the `expiry` of the `access_token` and saves it to keychain.
+Refreshes the `access_token`, then updates the keychain.
+(for more information please visit the following link -> https://datatracker.ietf.org/doc/html/rfc6749#section-6)
```
Comment (ipranay) [inline README.md]: I'm not sure if this adds much value (given that we have created the SDK to abstract this from the user)

Might be place these docs in the code base for some who is looking to see how this works.

File: README.md [hunk 7]
```diff
@@ -51,4 +55,87 @@ Fetches raw data and returns JSON,
...
+#### Swift Package Manager
+You can use [Swift Package Manager](https://swift.org/package-manager/) to install SwiftKeychainWrapper using Xcode:
+
+1. Open your project in Xcode
+
+2. Click "File" -> "Swift Packages" -> "Add Package Dependency..."
+
+3. Paste the following URL: https://github.com/deveronuas/sssdk.git
+
+4. Click "Next" -> "Next" -> "Finish"
```
Comment (ipranay) [inline README.md]:
```suggestion 
1. Open your project in Xcode
2. Click "File" -> "Add Packages..."
3. Paste the following URL: https://github.com/deveronuas/sssdk.git
4. Choose a `Dependency Rule` and finish
```

File: README.md [hunk 8]
```diff
@@ -51,4 +55,87 @@ Fetches raw data and returns JSON,
...
+
+## General Usage
+import SwiftUI
+import SimpleSalesforceSDK
+
+struct OAuthView: View {
+  let host = "your host url"
+  let clientId = "your client's id"
+  let clientSecret = "your client's secret"
+  let redirectUri = "your redirect uri"
```
Comment (ipranay) [inline README.md]: Let's use some convention for constants, like this?

```suggestion
  let HOST = "your host url"
  let CLIENT_ID = "your client's id"
  let CLIENT_SECRET = "your client's secret"
  let REDIRECT_URI = "your redirect uri"
```

File: README.md [hunk 9]
```diff
@@ -51,4 +55,87 @@ Fetches raw data and returns JSON,
...
+  var body: some View {
+    VStack{
+      Button{
+        launchLoginView.toggle()
+      }label: {
+        Text("Login")
+      }
+      .sheet(isPresented: $launchLoginView) {
+        try? SSSDK.shared.login()
+          .onOpenURL(perform: { url in
```
Comment (ipranay) [inline README.md]: not sure if this should be here... looks weird that this `onOpenURL` is attached to the login view

Comment (Asheshp23) [reply README.md]: I tried that we can use it anywhere but I can move it outside of that modal sheet.

File: .swiftpm/xcode/xcuserdata/apdev.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 10]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ipranay) [inline .swiftpm/xcode/xcuserdata/apdev.xcuserdatad/xcschemes/xcschememanagement.plist]: not sure if this file should be committed... looks like it is specific to your user on your mac

File: Sources/SimpleSalesforceSDK/LogInView.swift [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import SwiftUI
```
Comment (ipranay) [inline Sources/SimpleSalesforceSDK/LogInView.swift]: lets rename the file as well to `LoginView` with a small `i`

File: Sources/SimpleSalesforceSDK/SSSDK.swift [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+// custom runtime error
+public enum ConfigurationError : Error {
+  case runtimeError(String)
+}
+
+///
+/// SSSDK is a class to help to create the OAuth 2.0 user-agent flow,
+/// users authorize a mobile app to access data using an embedded browser.
+/// This OAuth 2.0 flow can replace the Salesforce's connected mobile SDK,
+/// by that you can directly utilize the salesforce API without the Salesforce's connected mobile SDK.
```
Comment (ipranay) [inline Sources/SimpleSalesforceSDK/SSSDK.swift:14]:
```suggestion
/// by that you can directly utilize the salesforce API without the Salesforce's connected mobile SDK.
/// 
/// Once authenticated, you can also use this class to make API calls to Salesforce.
```

File: Sources/SimpleSalesforceSDK/SSSDK.swift [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+///
+public class SSSDK {
+  /// Default SSSDK access
+  public static let shared = SSSDK()
```
Comment (ipranay) [inline Sources/SimpleSalesforceSDK/SSSDK.swift:19]: please create a private init if the intent is this class to be a singleton.

File: Sources/SimpleSalesforceSDK/SSSDK.swift [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+    
+    /// Requests the data using SOQL Query from the salesforce database.
+    /// - Parameters:
+    ///     - host: The Salesforce instance’s endpoint.
+    ///     - clientId: Client identifier for the OAuth 2.0 client.
+    ///     - clientSecret : Client Secret for the OAuth 2.0 client.
+    ///     - refreshToken : The refresh token issued to the client.
+    ///     - accessToken : The access token issued by the authorization server.
+    ///     - query : SOQL query to fetch the data.
+    /// - Returns: Data from the salesforce database.
```
Comment (ipranay) [inline Sources/SimpleSalesforceSDK/SSSDK.swift:136]: these should be in the WebService file above the function declaration, not where you call it.

**Review by ipranay — APPROVED** [review]
Some changes & cleanup I can make in another PR after this

**Pull request merged**